src/views/tables/MarketsCompactTable.tsx

```diff
@@ -134,31 +128,33 @@ export const MarketsCompactTable = ({
   );
 
   const sortedMarkets = useMemo(() => {
-    const sortingFunction = (marketA: MarketData, marketB: MarketData) => {
-      if (marketA.priceChange24HPercent == null && marketB.priceChange24HPercent == null) {
-        return 0;
-      }
+    if (sorting === MarketSorting.HIGHEST_CLOB_PAIR_ID) {
+      return filteredMarkets.sort((a, b) => b.clobPairId - a.clobPairId);
+    }
 
-      if (marketA.priceChange24HPercent == null) {
-        return 1;
-      }
+    if (sorting === MarketSorting.GAINERS || sorting === MarketSorting.LOSERS) {
+      const sortingFunction = (marketA: MarketData, marketB: MarketData) => {
+        if (marketA.priceChange24HPercent == null && marketB.priceChange24HPercent == null) {
+          return 0;
+        }
 
-      if (marketB.priceChange24HPercent == null) {
-        return -1;
-      }
+        if (marketA.priceChange24HPercent == null) {
+          return 1;
+        }
+
+        if (marketB.priceChange24HPercent == null) {
+          return -1;
+        }
 
-      return sorting === MarketSorting.GAINERS
-        ? marketB.priceChange24HPercent - marketA.priceChange24HPercent
-        : marketA.priceChange24HPercent - marketB.priceChange24HPercent;
-    };
+        return sorting === MarketSorting.GAINERS
+          ? marketB.priceChange24HPercent - marketA.priceChange24HPercent
+          : marketA.priceChange24HPercent - marketB.priceChange24HPercent;
+      };
 
-    if (sorting === MarketSorting.LOSERS) {
-      return filteredMarkets
-        .filter((market) => !!market.priceChange24HPercent && market.priceChange24HPercent < 0)
-        .sort(sortingFunction);
+      return filteredMarkets.sort(sortingFunction);
     }
```

Comment:
Simmplified some of the logic here. The filter not really necessary due to the sorting. Previously, gainers would not filter and just sort by the highest positive percent change. This means that it shows 5 markets even if all markets are down on the day. On the opposite side, losers would previously show no markets if all markets were up on the day.